core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1143,10 +1143,10 @@ private static void verifyChecksum(BufferedChecksumStreamInput in) throws IOExce
             }
             operation = Translog.Operation.readType(in);
             verifyChecksum(in);
+        } catch (TranslogCorruptedException e) {
+            throw e;
         } catch (EOFException e) {
             throw new TruncatedTranslogException("reached premature end of file, translog is truncated", e);
-        } catch (AssertionError | Exception e) {
-            throw new TranslogCorruptedException("translog corruption while reading from stream", e);
```

Comment:
@jpountz My thinking is that it's fine. It's either an `IOException` that is not an `EOFException`, or a general unknown exception. I do not think we should be wrapping these in translog corrupted exceptions because it might not actually be corrupt in the case of a general `IOException` or unknown exception. What do you think?